cmd/system-probe/api/client/client_others.go

```diff
@@ -3,34 +3,24 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016-present Datadog, Inc.
 
-//go:build !linux && !windows
+//go:build !unix && !windows
 
 package client
 
 import (
 	"context"
 	"net"
-	"net/http"
 	"time"
 )
 
 const (
-	netType = "tcp"
+	idleConnTimeout = 30 * time.Second
 )
 
-// newSystemProbeClient returns a http client configured to talk to the system-probe
-func newSystemProbeClient(socketPath string) *http.Client {
-	return &http.Client{
-		Timeout: 10 * time.Second,
-		Transport: &http.Transport{
-			MaxIdleConns:    2,
-			IdleConnTimeout: 30 * time.Second,
-			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
-				return net.Dial(netType, socketPath)
-			},
-			TLSHandshakeTimeout:   1 * time.Second,
-			ResponseHeaderTimeout: 5 * time.Second,
-			ExpectContinueTimeout: 50 * time.Millisecond,
-		},
+// DialContextFunc returns a function to be used in http.Transport.DialContext for connecting to system-probe.
+// The result will be OS-specific.
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏You could mention that DialContextFunc is not supported on this platform (as you did in `cmd/system-probe/api/server/listener_others.go`)